code/modules/mob/living/carbon/shock.dm

```diff
@@ -24,15 +23,25 @@
 
 	return src.traumatic_shock
 
+/mob/living/carbon/proc/getPainFromDam()
+	return src.getFireLoss() + src.getBruteLoss()
+
 // broken or ripped off organs will add quite a bit of pain
 /mob/living/carbon/human/updateshock()
 	..()
 	for(var/obj/item/organ/external/organ in organs)
-		if(organ && (organ.is_broken() || organ.open))
+		if(organ && (organ.is_broken() || (!BP_IS_ROBOTIC(organ) && organ.open)))
 			traumatic_shock += 30
 
 	return traumatic_shock
 
+/mob/living/carbon/human/getPainFromDam()
+	var/value = 0
+	for(var/obj/item/organ/external/organ in organs)
+		value += organ.burn_dam
+		value += organ.brute_dam
+		value *= (get_specific_organ_efficiency(OP_NERVE, organ.organ_tag)/100)
```

Comment:
```suggestion
		value *= max((get_specific_organ_efficiency(OP_NERVE, organ.organ_tag)/100), 0.5)
```
Would cap it at a 0.5 modifier, otherwise people would just take the 33% effciency hit to feel no pain at all.

Reply:
And here I thought it was a 100% efficiency hit